.github/workflows/labeler.yml

```diff
@@ -12,6 +13,28 @@ jobs:
       pull-requests: write  # for actions/labeler to add labels to PRs
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/labeler@main
+      - name: Use Git Actions labeler
+        uses: actions/labeler@main
         with:
           repo-token: "${{ secrets.GITHUB_TOKEN }}"
+
+      - name: Fetch pull request base branch
+        id: pr_info
+        run: |
+          BASE_BRANCH="${{ github.event.pull_request.base.ref }}"
+          if [ "$BASE_BRANCH" == "main" ]; then
+            echo "::set-output name=base_branch_name::inspec-6"
```

Comment:
This will not age well - we'll have to update this every time we have a new major version. I'm not sure how to get out of that, though. At least adding a comment would be good.

Reply:
Maybe I can read the version file content, and fetch the major version from it and append "inspec-#{VERSION}", when the base branch is main.

And for the rest, If the base branch starts with `inspec-` then just directly pick the base branch. This way it gets generic but it takes away our safe checks `[ "$BASE_BRANCH" == "inspec-4" ] || [ "$BASE_BRANCH" == "inspec-5" ]`. 
By this I mean, if by any chance someone creates a dummy branch with `inspec-` then it will create a label for that. It could be `inspec-random-branch-name` or `inspec-100` etc..

Reply:
Will have to check though if reading file content is possible or not. I know matching is possible, will play around with reading content though.